File: Challenges/Coding World Cup/2023/Day 27 - PAK vs BAN - Bait and switch.java
import java.util.Arrays;

public class Solution {
    public static long baitAndSwitch(int n, int[] a, int k, int m) {
        int[] diff = new int[n];
        for (int i = 0; i < n; i++) {
            diff[i] = Math.abs(a[i] - m);
        }

        Arrays.sort(diff);

        long s = 0;
        for (int i = 0; i < k; i++) {
            s += diff[n - 1 - i];
        }

        return s;
    }
}

File: Challenges/Coding World Cup/2023/Day 27 - PAK vs BAN - Bait and switch.cpp
long long baitAndSwitch(int n, vector<int> &a, int k, int m)
{
    // Write your code here.
    vector<int> diff;
    for (int i = 0; i < n; i++)
    {
        diff.push_back(abs(a[i] - m));
    }

    sort(diff.begin(), diff.end());

    long long s = 0;
    for (int i = 0; i < k; i++)
    {
        s += diff[n - 1 - i];
    }

    return s;
}

